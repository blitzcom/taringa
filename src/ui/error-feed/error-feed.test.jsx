import renderer from 'react-test-renderer';

import ErrorFeed from './error-feed';

describe('ui/error-feed', () => {
  def('onRetry', () => jest.fn());
  def('tree', () => renderer.create(<ErrorFeed onRetry={$onRetry} />));
  def('subject', () => $tree.toJSON());

  it('renders error feed', () => {
    expect($subject).toMatchSnapshot();
  });

  it('calls on retry', () => {
    $tree.root.findByType('button').props.onClick();
    expect($onRetry).toHaveBeenCalled();
  });
});
