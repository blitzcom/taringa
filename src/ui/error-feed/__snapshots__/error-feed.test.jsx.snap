// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/error-feed renders error feed 1`] = `
<div
  className="wrapper"
>
  <h2
    className="title"
  >
    We have a little problem
  </h2>
  <p
    className="description"
  >
    Can't fetch feed. Please make sure you are connected to the internet.
  </p>
  <button
    className="button"
    onClick={[MockFunction]}
  >
    Try again
  </button>
</div>
`;
